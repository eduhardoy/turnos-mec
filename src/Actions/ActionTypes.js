export const actionTypes = {
    ADD_USER: "ADD_USER",
    DELETE_USER: "DELETE_USER",
    ADD_STEP_ACTIVE: "ADD_STEP_ACTIVE",
    SUBSTRACT_STEP_ACTIVE: "SUBSTRACT_STEP_ACTIVE",
    ADD_SHIFT: "ADD_SHIFT",
    DELETE_SHIFT: "DELETE_SHIFT"
}