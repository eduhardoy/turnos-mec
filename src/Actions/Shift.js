import { actionTypes } from './ActionTypes'


export const addShift = payload => {
    return {
        type: actionTypes.ADD_SHIFT,
        payload
    }
}

export const deleteShift = () => {
    return {
        type: actionTypes.DELETE_SHIFT,
    }
}
