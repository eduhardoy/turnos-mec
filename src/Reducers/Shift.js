import { actionTypes } from '../Actions/ActionTypes'

export default (state = {}, action) => {
    switch (action.type) {
        case actionTypes.ADD_SHIFT:
            return action.payload
        case actionTypes.DELETE_SHIFT:
            return action.payload
        default:
            return state
    }
}