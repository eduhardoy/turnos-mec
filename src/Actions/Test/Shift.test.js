import * as actions from '../Shift'
import * as types from '../ActionTypes'

describe('addShift returns action', () => {
    it('return an action', () => {
        const payload = {direccion: 'NivelSecundario'}
        expect(actions.addShift(payload)).toMatchSnapshot()
    })
})