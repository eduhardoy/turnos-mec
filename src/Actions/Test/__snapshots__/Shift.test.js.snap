// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addShift returns action return an action 1`] = `
Object {
  "payload": Object {
    "direccion": "NivelSecundario",
  },
  "type": "ADD_SHIFT",
}
`;
